File: spotify.client/src/models/UserData.js
// User data model for persistent storage
export class UserData {
  constructor(userId) {
    this.userId = userId;
    this.likedSongs = [];
    this.playlists = [];
  }

  save() {
    localStorage.setItem(`userData_${this.userId}`, JSON.stringify({
      likedSongs: this.likedSongs,
      playlists: this.playlists
    }));
  }

  static load(userId) {
    const userData = new UserData(userId);
    const saved = localStorage.getItem(`userData_${userId}`);
    if (saved) {
      const data = JSON.parse(saved);
      userData.likedSongs = data.likedSongs || [];
      userData.playlists = data.playlists || [];
    }
    return userData;
  }

  likeSong(song) {
    if (!this.likedSongs.find(s => s.title === song.title)) {
      this.likedSongs.push(song);
      this.save();
    }
  }

  unlikeSong(song) {
    this.likedSongs = this.likedSongs.filter(s => s.title !== song.title);
    this.save();
  }

  isSongLiked(song) {
    return this.likedSongs.some(s => s.title === song.title);
  }

  createPlaylist(name) {
    const playlist = {
      id: Date.now(),
      name: name,
      songs: [],
      createdAt: new Date().toISOString()
    };
    this.playlists.push(playlist);
    this.save();
    return playlist;
  }

  addToPlaylist(playlistId, song) {
    const playlist = this.playlists.find(p => p.id === playlistId);
    if (playlist && !playlist.songs.find(s => s.title === song.title)) {
      playlist.songs.push(song);
      this.save();
    }
  }

  deletePlaylist(playlistId) {
    this.playlists = this.playlists.filter(p => p.id !== playlistId);
    this.save();
  }
}

File: Spotify.Server/Models/UserData.cs
namespace Spotify.Server.Models
{
    public class UserData
    {
        public string UserId { get; set; } = string.Empty;
        public List<Song> LikedSongs { get; set; } = new();
        public List<Playlist> Playlists { get; set; } = new();
    }

    public class Song
    {
        public string Title { get; set; } = string.Empty;
        public string Artist { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
    }

    public class Playlist
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public List<Song> Songs { get; set; } = new();
        public DateTime CreatedAt { get; set; }
    }
}